src/include/network/protocol_interpreter.h

```diff
@@ -90,7 +90,7 @@ class ProtocolInterpreter {
    * @param in The ReadBuffer to read input from
    * @return whether the packet header is valid or not
    */
-  bool TryReadPacketHeader(const common::ManagedPointer<ReadBuffer> in) {
+  virtual bool TryReadPacketHeader(const common::ManagedPointer<ReadBuffer> in) {
```

Comment:
Is there a reason to make this virtual?

Reply:
After looking at the new implementations, I do not think this should be made virtual. I think `TryBuildPacket`'s override is identical to the original, and `TryReadPacketHeader`'s override is almost identical except for a copy of the buffer (which could happen elsewhere).